vscode-ext/src/util.ts

```diff
@@ -0,0 +1,20 @@
+import * as vscode from 'vscode';
+
+/// Key used to store the last selected target in the persistent state.
+export const LAST_TARGET_KEY = "mirrord-last-target";
+
+/// Option in the target selector that represents no target.
+export const TARGETLESS_TARGET_NAME = "No Target (\"targetless\")";
+
+// Display error message with help
+export function mirrordFailure(err: any) {
```

Comment:
I'd merge this entire file into api.ts, it's only used there.

Reply:
its also used by the extension.ts file, however i believe it makes sense to keep it with the API, we don't really have any "utility functions" as such anyway, fixed!